### ISSUE TITLE
Crash when writing a large json to the workplace

### ISSUE THREAD

**Issue by khvorov45** [slynch8/10x#1815]
Workplace

```
crash10x
├── build.bat
├── crash.c
├── crash10x.10x
```

`crash.c` writes `crash.json` to the workplace

```c
#include "Windows.h"

#define assert(x) do { if (x) {} else { __debugbreak(); } } while (0)

#define STR(x) (Str) { x, sizeof(x) - 1 }
typedef struct Str {
	char* ptr;
	intptr_t len;
} Str;

typedef struct StrBuilder {
	char* ptr;
	intptr_t len;
	intptr_t cap;
} StrBuilder;

static void addSegment(StrBuilder* builder, Str segment) {
	assert(builder->cap - builder->len >= segment.len);
	CopyMemory(builder->ptr + builder->len, segment.ptr, segment.len);
	builder->len += segment.len;
}

int main() {
	intptr_t bytes = 1 * 1024 * 1024 * 1024;
	void* mem = VirtualAlloc(0, (SIZE_T)bytes, MEM_RESERVE | MEM_COMMIT, PAGE_READWRITE);
	assert(mem);

	Str out = {};
	{
		intptr_t pairCount = 1000000;

		StrBuilder builder = { mem, 0, bytes };
		addSegment(&builder, STR("{\"pairs\":[\n"));
		for (intptr_t ind = 0; ind < pairCount; ind++) {
			addSegment(&builder, STR("    {\"x0\":1.0000000000000000, \"x1\":1.0000000000000000, \"y0\":1.0000000000000000, \"y1\":1.0000000000000000}"));
			if (ind < pairCount - 1) {
				addSegment(&builder, STR(","));
			}
			addSegment(&builder, STR("\n"));
		}

		addSegment(&builder, STR("]}"));
		out = (Str) { builder.ptr, builder.len };
	}

	HANDLE file = CreateFileA("crash.json", GENERIC_WRITE, 0, 0, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, 0);
	assert(file != INVALID_HANDLE_VALUE);

	DWORD written = 0;
	WriteFile(file, out.ptr, out.len, &written, 0);
	assert(written == out.len);

	return 0;
}
```
Set up `build.bat` as the build/run command
```
clang crash.c -g -o crash.exe && crash.exe
```
It probably won't crash the first time because crash.json won't be in the workplace. Once that file makes it into the workplace (after "sync folder") running the build command crashes 10x every time (for me at least).

Attached .10x and a crash dump (note appended file extensions). I also uploaded that dump through the website.

[crash10x.10x.txt](https://github.com/slynch8/10x/files/11765178/crash10x.10x.txt)
[10x_crash_2023_6_16_11_45_11.dmp.gzip.zip](https://github.com/slynch8/10x/files/11765222/10x_crash_2023_6_16_11_45_11.dmp.gzip.zip)

**Comment by slynch8** [comment]
Hi, I've getting lots of compile and link errors trying to build the exe, could you post the crash.json file please?

**Comment by slynch8** [comment]
ignore that. I've built it with MSVC instead. But it's not crashing for me. It's possible that I fixed this a few builds back, does it still crash for you?

**Comment by khvorov45** [comment]
Still crashing on 1.0.48
It's the regeneration of crash.json that does it. I can reduce build.bat to just `crash.exe` and it crashes on build.

**Comment by slynch8** [comment]
fixed (1.0.49)
